python/pip_install/pip_compile.py

```diff
@@ -73,7 +73,15 @@
         cli()
         print("cli() should exit", file=sys.stderr)
         sys.exit(1)
-    except SystemExit:
+    except SystemExit as e:
+        if e.code == 2:
```

Comment:
This improves things for sure, but still leaves us open to cases where `e.code not in {0, 2}`. Looking at the piptools source, it only ever exits 0 or 2, but my preference would be to check for the unexpected cases and `exit(1)` instead of incorrectly succeeding .

```python
if e.code == 2:
   # handle 2
   # ...
   sys.exit(1)
elif e.code == 0:
   # handle 0
   # ...
   sys.exit(0)
else:
   # handle unexpected
   # ...
   sys.exit(1)
```

Reply:
I totally agree. I have fixed it that way.